plugins/jobtype/src/azkaban/jobtype/HadoopSecureSparkWrapper.java

```diff
@@ -0,0 +1,157 @@
+
+    if (shouldProxy(jobProps)) {
+      UserGroupInformation proxyUser = null;
+      String userToProxy = jobProps.getProperty("user.to.proxy");
+      if (securityEnabled) {
+        proxyUser = createSecurityEnabledProxyUser(userToProxy);
+      } else {
+        proxyUser = UserGroupInformation.createRemoteUser(userToProxy);
+      }
+
+      logger.info("Proxied as user " + userToProxy);
+
+      proxyUser.doAs(new PrivilegedExceptionAction<Void>() {
+        @Override
+        public Void run() throws Exception {
+          runSpark(args);
+          return null;
+        }
+      });
+
+    } else {
+      logger.info("Not proxying. ");
+      runSpark(args);
+    }
+  }
+
+  /**
+   * Actually calling the spark-submit command
+   * 
+   * @param args
+   */
+  private static void runSpark(String[] args) {
+
+    logger.info("args: " + Arrays.toString(args));
+
+    org.apache.spark.deploy.SparkSubmit$.MODULE$.main(args);
+  }
+
+  /**
+   * Perform all the magic required to get the proxyUser in a securitized grid
+   * 
+   * @param userToProxy
+   * @return
+   * @throws IOException
+   */
+  private static UserGroupInformation createSecurityEnabledProxyUser(String userToProxy)
+          throws IOException {
+    UserGroupInformation proxyUser;
+    String filelocation = System.getenv(HADOOP_TOKEN_FILE_LOCATION);
+    if (filelocation == null) {
+      throw new RuntimeException("hadoop token information not set.");
+    }
+    if (!new File(filelocation).exists()) {
+      throw new RuntimeException("hadoop token file doesn't exist.");
+    }
+
+    logger.info("Found token file " + filelocation);
+
+    logger.info("Setting " + HadoopSecurityManager.MAPREDUCE_JOB_CREDENTIALS_BINARY + " to "
+            + filelocation);
+    System.setProperty(HadoopSecurityManager.MAPREDUCE_JOB_CREDENTIALS_BINARY, filelocation);
+
+    UserGroupInformation loginUser = null;
+
+    loginUser = UserGroupInformation.getLoginUser();
+    logger.info("Current logged in user is " + loginUser.getUserName());
+
+    logger.info("Creating proxy user.");
+    proxyUser = UserGroupInformation.createProxyUser(userToProxy, loginUser);
+
+    for (Token<?> token : loginUser.getTokens()) {
+      proxyUser.addToken(token);
+    }
+    return proxyUser;
+  }
+
+  /**
+   * Loading the properties file, which is a combination of the jobProps file and sysProps file
+   * 
+   * @return
+   * @throws IOException
+   * @throws FileNotFoundException
+   */
+  private static Properties loadCombinedProps() throws IOException, FileNotFoundException {
+    String propsFile = System.getenv(ProcessJob.JOB_PROP_ENV);
+    Properties props = new Properties();
+    props.load(new BufferedReader(new FileReader(propsFile)));
+    return props;
+  }
+
+  private static boolean shouldProxy(Properties props) {
+    String shouldProxy = props.getProperty(HadoopSecurityManager.ENABLE_PROXYING);
+
+    return shouldProxy != null && shouldProxy.equals("true");
```

Comment:
This is by default taking java.util.Properties, not azkaban.Props, so doesn't have that method.  Though we can force people to pass in Props if we want